### ISSUE TITLE
[Dwarf Manipulator] Building Designer does not activate Architect

### ISSUE THREAD

**Issue by boxofrox** [DFHack/dfhack#540]
Using 0.40.24 of Dwarf Fortress and DFHack.  I noticed that no dwarf would build my bridges, and the bridge jobs complained an Architect was needed.  None of my dwarfs had any experience in Architect when I assigned one as Architect.

**Steps to Reproduce**
1. Open Dwarf Manipulator. (`u`, `l`)
2. Enable Building Designer for any dwarf, A, not already assigned.  Remember this dwarf's name.
3. Return to main menu.
4. Open profession for dwarf A.  (`u`, select dwarf A, `z`, `p`, `l`, select *Other Jobs*, `Enter`)
5. Observe that Architect is not enabled.

**Expected Behavior**
When Building Designer is enabled, Architect is also enabled.

**Comment by fricy** [comment]
Can't reproduce. [OSX].40.24-r2
check if you have autolabor running

**Comment by boxofrox** [comment]
Autolabor is disabled.

Enabled plugins:
* autodump-destroy-here
* autodump-destroy-item
* command-prompt
* copystock
* digl
* diglx
* digtype
* digv
* digvx
* filltraffic
* follow
* job-duplicate
* job-material
* liquids-here
* loadstock
* savestock
* sort-items
* spotclean
* tidlers
* twaterlvl

Running on Archlinux.  Built DFHack 0.40.24-r2 using AUR(arch user repository) and installed that and Dwarf Fortress 0.40.24 through the Arch package manager into /opt/dwarffortress.

**Comment by boxofrox** [comment]
I'll try downloading dfhack 0.40.24-r2 and dwarffortress 0.40.24 without using the Arch packages to verify the packages aren't the cause.

**Comment by boxofrox** [comment]
Meh.  I can no longer reproduce this with the versions installed through Arch package manager.  I dunno whether it was the particular dwarf I selected, or that I didn't use default party when I embarked, or the dwarf was the first ever assigned to Architect, or what.  Gonna close this until I can find more reliable steps to reproduce.

Thanks for pointing out autolabor.  I'm new to DFHack and never suspected it might be on (even though it wasn't).
